src/core_modules/capture-core/components/Breadcrumbs/EnrollmentBreadcrumb/hooks/useWorkingListLabel.js

```diff
@@ -0,0 +1,60 @@
+// @flow
+import i18n from '@dhis2/d2-i18n';
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+
+type Props = {
+    programId: string,
+    displayFrontPageList: boolean,
+    trackedEntityName?: string,
+}
+
+const DefaultFilterLabels = {
+    active: i18n.t('Active enrollments'),
+    complete: i18n.t('Completed enrollments'),
+    cancelled: i18n.t('Cancelled enrollments'),
+};
+
+export const useWorkingListLabel = ({
+    programId,
+    trackedEntityName,
+    displayFrontPageList,
+}: Props) => {
+    const workingListTemplates = useSelector(({ workingListsTemplates }) => workingListsTemplates?.teiList);
+    const workingListProgramId = useSelector(({ workingListsContext }) => workingListsContext?.teiList?.programIdView);
```

Comment:
For me it looks like the two useSelector calls are rendered 10 times each whenever navigating to the dashboard for a TEI. Should this behavior be optimized?"

Reply:
It looks like `workingListProgramId` triggers one render and `workingListTemplates` triggers two. (One of each is batched, so they should only really trigger one rerender. Did you have any other experience, @henrikmv? Remember, if you want to see it the value triggers a rerender, you will have to wrap the `console.log` inside a useEffect.